lib/reporters.js

```diff
@@ -27,38 +27,37 @@ class NullReporter {
 
 /**
  * @private
- * A custom HTTP implementation for Datadog that retries failed requests.
- * Datadog has retries built in, but they don't handle network errors (just
- * HTTP errors), and we want to retry in both cases. This inherits from the
- * built-in HTTP library since we want to use the same fetch implementation
- * Datadog uses instead of adding another dependency.
+ * Manages HTTP requests and associated retry/error handling logic.
  */
-class RetryHttp extends datadogApiClient.client.IsomorphicFetchHttpLibrary {
-    constructor(options = {}) {
-        super(options);
-
-        // HACK: ensure enableRetry is always `false` so the base class logic
-        // does not actually retry (since we manage retries here).
-        Object.defineProperty(this, 'enableRetry', {
-            get () { return false; },
-            set () {},
-        });
+class HttpApi {
```

Comment:
Probably want to either merge this into `DatadogReporter` or pull some more logic from it (e.g. base URL stuff) down into this class.